plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -633,57 +624,89 @@ def finalize(self):
                 add_metadata_file_metadata('updateinfo', self.context.metadata_file_path,
                                            self.context.checksum)
 
-    def _get_pkglists_to_publish(self, erratum_unit):
+    def _get_repo_unit_nevra(self, repo_id):
+        """
+        Return a list of NEVRA tuples for units in a single repo referenced by the given errata.
+
+        Pulp errata units combine the known packages from all synced repos. Given an errata unit
+        and a repo, return a set of NEVRA tuples that can be used to filter out packages not
+        linked to that repo when generating a repo's updateinfo XML file.
+
+        :param repo_id: Repository ID for which to return the set of nevra
+        :type repo_id: str
+
+        :return: a set of NEVRA dicts for units in a single repo referenced by the given errata
+        :rtype: set of pulp_rpm.plugins.models.NEVRA tuples
+        """
+        repo_unit_nevra = set()
+        querysets = repo_controller.get_unit_model_querysets(repo_id, models.RPM)
+        nevra_scalars = itertools.chain(*[q.scalar(*models.NEVRA._fields) for q in querysets])
+        for scalar in nevra_scalars:
+            repo_unit_nevra.add(models.NEVRA(*scalar))
+        return repo_unit_nevra
+
+    def _get_pkglist_to_publish(self, erratum_unit):
         """
         Make a list of packages which will be listed in the published erratum.
 
         It is possible to filter packages only if `repo_unit_nevra` is available,
```

Comment:
I would also add something about filtering out packages which are not in a repo, otherwise from the docstring it seems like the only thing this method does is a merge of pkglists into one.